pkg/controllers/vdb/createarchive_reconciler.go

```diff
@@ -0,0 +1,203 @@
+	vapi "github.com/vertica/vertica-kubernetes/api/v1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/createarchive"
+	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/saverestorepoint"
+	"github.com/vertica/vertica-kubernetes/pkg/vdbstatus"
+	"github.com/vertica/vertica-kubernetes/pkg/version"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreateArchiveReconciler will create archive and save restore point if triggered
+type CreateArchiveReconciler struct {
+	VRec *VerticaDBReconciler
+	Vdb  *vapi.VerticaDB
+	Log  logr.Logger
+	client.Client
+	Dispatcher     vadmin.Dispatcher
+	PFacts         *PodFacts
+	OriginalPFacts *PodFacts
+	InitiatorIP    string // The IP of the pod that we run vclusterOps from
+}
+
+func MakeCreateArchiveReconciler(r *VerticaDBReconciler, vdb *vapi.VerticaDB, log logr.Logger,
+	pfacts *PodFacts, dispatcher vadmin.Dispatcher, cli client.Client) controllers.ReconcileActor {
+	pfactsForMainCluster := pfacts.Copy(vapi.MainCluster)
+	return &CreateArchiveReconciler{
+		VRec:           r,
+		Log:            log.WithName("CreateArchiveReconciler"),
+		Vdb:            vdb,
+		Client:         cli,
+		Dispatcher:     dispatcher,
+		PFacts:         &pfactsForMainCluster,
+		OriginalPFacts: pfacts,
+	}
+}
+
+// Reconcile will create an archive if the status condition indicates true
+// And will save restore point to the created arcihve if restorePoint.archive value
+// is provided in the CRD spec
+func (c *CreateArchiveReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	err := c.PFacts.Collect(ctx, c.Vdb)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// No-op if no database exists
+	if !c.PFacts.doesDBExist() {
+		return ctrl.Result{}, nil
+	}
+
+	hostIP, ok := c.PFacts.FindFirstUpPodIP(true, "")
+	if !ok {
+		// If no running pod found, then there is nothing to do and we can just continue on
+		return ctrl.Result{}, nil
+	}
+	// Ensure vertica version
+	var vinf *version.Info
+	if !vinf.IsEqualOrNewer(vapi.SaveRestorePointNMAOpsMinVersion) {
+		c.VRec.Eventf(c.Vdb, corev1.EventTypeWarning, events.UnsupportedVerticaVersion,
+			"The Vertica version %q doesn't support create restore points. The minimum version supported is %s.",
+			vinf.VdbVer, vapi.SaveRestorePointNMAOpsMinVersion)
+		err := vdbstatus.UpdateCondition(ctx, c.VRec.Client, c.Vdb,
+			vapi.MakeCondition(vapi.SaveRestorePointsNeeded,
+				metav1.ConditionFalse, "IncompatibleDB"),
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Only proceed if the SaveRestorePointsNeeded status condition is set to true.
+	if c.Vdb.IsStatusConditionTrue(vapi.SaveRestorePointsNeeded) {
```

Comment:
This must be the first thing you check.

Reply:
Changed the order in https://github.com/vertica/vertica-kubernetes/pull/918/commits/923003d736d7f66cd58f43977f4e73647400bca7